src/backend/PortabilityService.ConfigurationSource/ConfigurationSource.cs

```diff
@@ -0,0 +1,89 @@
+    {
+        private const string DefaultConfigurationServiceUrl = "http://localhost:1443";
+        internal const string PortabilityServiceConfigurationRoot = "PortabilityServiceSettings";
+
+        public PortabilityServiceConfigurationSource(string configurationSection = PortabilityServiceConfigurationRoot)
+            : this(GetConfigurationServiceUrlFromEnvironment() ?? new Uri(DefaultConfigurationServiceUrl), configurationSection) { }
+
+        public PortabilityServiceConfigurationSource(Uri configurationServiceUrl, string configurationSection = PortabilityServiceConfigurationRoot)
+            : this(CreateHttpClient(configurationServiceUrl ?? throw new ArgumentNullException(nameof(configurationServiceUrl))), configurationSection ?? throw new ArgumentNullException(nameof(configurationSection))) { }
+
+        public PortabilityServiceConfigurationSource(HttpClient httpClient, string configurationSection = PortabilityServiceConfigurationRoot)
+        {
+            InitialData = GetConfigurationSectionSettings(configurationSection ?? throw new ArgumentNullException(nameof(configurationSection)), httpClient ?? throw new ArgumentNullException(nameof(httpClient))).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Gets the Portability configuration service URL from the environment
+        /// </summary>
+        /// <returns></returns>
+        private static Uri GetConfigurationServiceUrlFromEnvironment()
+        {
+            var urlFromEnvironment = Environment.GetEnvironmentVariable("PortabilityConfigurationServiceUrl");
+
+            if (string.IsNullOrEmpty(urlFromEnvironment))
+            {
+                return null;
+            }
+            else
+            {
+                return new Uri(urlFromEnvironment);
+            }
+        }
+
+        /// <summary>
+        /// Creates an HTTPClient with the appsettings.json Url
+        /// </summary>
+        private static HttpClient CreateHttpClient(Uri configurationServiceUrl) => new HttpClient() { BaseAddress = configurationServiceUrl };
+
+        /// <summary>
+        /// Gets the configuration settings for the passed in section
+        /// </summary>
+        private static async Task<IEnumerable<KeyValuePair<string, string>>> GetConfigurationSectionSettings(string configurationSection, HttpClient httpClient)
+            => await GetJsonDataAsync<IEnumerable<KeyValuePair<string, string>>>($"/api/Configuration/sectionsettingslist/{configurationSection}", httpClient);
+
+        /// <summary>
+        /// Returns Json from the WebAPI call with passed in URI
+        /// </summary>
+        private static async Task<T> GetJsonDataAsync<T>(string uri, HttpClient httpClient)
+        {
+            using (var request = new HttpRequestMessage(HttpMethod.Get, uri))
+            {
+                request.Headers.Add("Accept", "application/json");
+                return await GetJsonDataAsync<T>(request, httpClient);
```

Comment:
Add: `.ConfigureAwait(false)`

Reply:
Added

Reply:
We should add the fxcop analyzers that check for this (not in this PR, but could be a task or something)